bin/webpack-dev-server

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env ruby
```

Comment:
Hmm, so my accessing this at `/public/lib/.bin/webpack-dev-server` seems wrong, or maybe it's correct and we don't also need it here, esp if this copy of it would go out of date if we upgraded webpack? Maybe better to rely on the one in our /public/lib/ folder?

However, i think we are seeing some of these errors because we have an atypical location for our yarn dependencies, in public/lib instead of node_modules. I see this error when i try running `webpack-dev-server`:

```
gitpod /workspace/plots2 $ bin/webpack-dev-server 
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'webpack-cli/bin/config-yargs'
Require stack:
- /workspace/plots2/node_modules/webpack-dev-server/bin/webpack-dev-server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/workspace/plots2/node_modules/webpack-dev-server/bin/webpack-dev-server.js:65:1)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/workspace/plots2/node_modules/webpack-dev-server/bin/webpack-dev-server.js'
  ]
}
```

Reply:
@jywarren Thanks for doing so much work trying to set this up. It's nice to see some progress, although frustrating we can't get it working yet.

My understanding is that `webpack-dev-server` is just supposed to be used in development, which I guess GitHub CI & GitPod don't count as. So it does seem like that is more of a workaround than an actual solution.

Locally, I have a public/lib folder, and it's filled with dependencies. Is that normal? Are all the yarn dependencies supposed to be consolidated in `node_modules`? I don't know! This is just how the setup happened after `rails generate`.

I'm going to run `rails test functional` to see what happens.

There are also alternatives to the `react-rails` Gem that could work. Even after that swap, all the React components I write can probably just plug into that as written.

Reply:
<img width="613" alt="Screen Shot 2021-02-23 at 2 49 08 PM" src="https://user-images.githubusercontent.com/4361605/108918529-60afe480-75e6-11eb-92ec-b930dc6c4080.png">

Still getting this when I open GitPod on this PR

Reply:
Functional tests seem to be running fine locally BTW, I don't get the `manifest.json` errors that appear in CI

Reply:
> in development, which I guess GitHub CI & GitPod don't count as. So it does seem like that is more of a workaround than an actual solution.

GitPod should be just like local development, so development mode, but you have a very good point on GitHub CI -- it's in test mode! We should search webpacker docs for how to get it running in test mode...

> Locally, I have a public/lib folder, and it's filled with dependencies. Is that normal? Are all the yarn dependencies supposed to be consolidated in node_modules? 

Yes, that's normal for plots2 - we were probably ahead of the curve on switching to Yarn so we have a custom location. We could consider switching to a more conventional location but only if it's essential, as that's a pretty big change just to get React running. I mean, we will probably have to do this for Rails 6 at some point in any case...

No worries here, this all seems like a pretty normal amount of difficulty for integrating a system like React+Webpack! Keep up the great work and i'm sure we'll crack it!